src/community/cog/src/test/java/org/geoserver/rest/catalog/CogRemoteHarvestTest.java

```diff
@@ -0,0 +1,118 @@
+/* (c) 2020 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.rest.catalog;
+
+import static org.custommonkey.xmlunit.XMLAssert.assertXpathEvaluatesTo;
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.util.HashMap;
+import java.util.Map;
+import org.custommonkey.xmlunit.SimpleNamespaceContext;
+import org.custommonkey.xmlunit.XMLAssert;
+import org.custommonkey.xmlunit.XMLUnit;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.rest.RestBaseController;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.w3c.dom.Document;
+
+public class CogRemoteHarvestTest extends CatalogRESTTestSupport {
```

Comment:
This test talks to a remote resource that might not be available (e.g., machine not connected to internet), or the image in question might be eventually removed, or moved. Is it possible to make it an online test, so that it runs only when a  given profile is used, or at least check if the remote file is there?
Could be done with an Assume I suppose, and code similar to this to verify the remote file is at least reachable: https://github.com/geoserver/geoserver/blob/master/src/web/core/src/test/java/org/geoserver/web/GeoServerWicketOnlineTestSupport.java#L19
I was also considering calling this "CogRemoteHarvestOnlineTest", but unlike GeoTools, GeoServer does not have online test skipping, AFAIK.